ui/src/core_components/content/CoreChat.vue

```diff
@@ -0,0 +1,519 @@
+									<div
+										v-if="action.subheading"
+										class="subheading"
+									>
+										{{ action.subheading }}
+									</div>
+									<h3 class="name">{{ action.name }}</h3>
+									<div v-if="action.desc" class="desc">
+										{{ action.desc }}
+									</div>
+								</button>
+							</div>
+						</template>
+					</div>
+					<div
+						v-if="message.date"
+						class="time"
+						:title="getFormattedDate(message.date, false)"
+					>
+						{{ getFormattedDate(message.date, true) }}
+					</div>
+				</div>
+			</div>
+		</div>
+		<div class="inputArea">
+			<textarea
+				v-model="outgoingMessage"
+				placeholder="Write something..."
+				@keydown.prevent.enter="handleMessageSent"
+			></textarea>
+			<button @click="handleMessageSent">
+				<i class="ri-send-plane-line"></i>
+			</button>
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import LoadingSymbol from "../../renderer/LoadingSymbol.vue";
+import BaseMarkdown from "../base/BaseMarkdown.vue";
+import { FieldCategory, FieldType } from "../../streamsyncTypes";
+import {
+	buttonColor,
+	buttonTextColor,
+	containerBackgroundColor,
+	cssClasses,
+	primaryTextColor,
+	secondaryTextColor,
+	separatorColor,
+} from "../../renderer/sharedStyleFields";
+import { onMounted } from "vue";
+import { onBeforeUnmount } from "vue";
+
+const description = "A chat component to build human-to-AI interactions.";
+
+const docs = `
+Connect it to an LLM by handling the \`ss-chat-message\` event, which is triggered every time the user sends a message. When the response is ready, return it.
+
+You can add \`actions\` to your response, which are buttons that trigger the \`ss-chat-action-click\`.
+
+See the stubs for more details.`.trim();
+
+const chatMessageStub = `
+def handle_message_simple(payload):
+    query = payload
+
+    if query == "Hello":
+
+		# You can simply return a string
+
+        return "Hello, human."
+    elif query == "Surprise me":
+
+		# Or you can return a dict with actions, which are buttons
+		# added to the conversation
+
+        return {
+            "text": "I can help you with that.",
+            "actions": [{
+            	"subheading": "Resource",
+            	"name": "Surprise",
+            	"desc": "Click to be surprised",
+            	"data": "change_title" 
+        	}]
+        }
+    else:
+        return "I don't know"
+`.trim();
+
+const chatActionClickStub = `
+def handle_action_simple(payload, state):
+    
+    # payload contains the "data" property of the action 
+    
+    if payload == "change_title":
+        state["app_title"] = "Surprise!"
+        state["app_background_color"] = "red"
+    
+    # A message can be added to the chat
+    
+    return "Hope you're surprised."
+`.trim();
+
+export default {
+	streamsync: {
+		name: "Chat",
+		description,
+		docs,
+		category: "Content",
+		fields: {
+			incomingInitials: {
+				name: "Incoming initials",
+				default: "AI",
+				type: FieldType.Text,
+			},
```

Comment:
In a normal chat, absolutely. But since this is in fact a component to create chatbots, it's expected that it'll always be the same entity. "AI" or for example if you want to create a "Coding Companion", "CC".

Not requiring this also allows you to simply pass a string for the response which is nice.

Perhaps the question is, should this component be named "chatbot" instead?